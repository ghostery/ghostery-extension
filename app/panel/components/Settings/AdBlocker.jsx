/**
 * Ad Blocker Settings Component
 *
 * Ghostery Browser Extension
 * https://www.ghostery.com/
 *
 * Copyright 2020 Ghostery, Inc. All rights reserved.
 *
 * This Source Code Form is subject to the terms of the Mozilla Public
 * License, v. 2.0. If a copy of the MPL was not distributed with this
 * file, You can obtain one at http://mozilla.org/MPL/2.0
 */

import React from 'react';
import PropTypes from 'prop-types';
import { RadioButtonGroup } from '../BuildingBlocks';

/**
 * @class Implement Ad Blocker Settings subview as a React component.
 * The view opens from the left-side menu of the main Settings view.
 * It allows the user to choose their Ad Blocker filter lists.
 * @memberOf SettingsComponents
 */
const AdBlocker = (props) => {
	const { settingsData } = props;

	const handleListSelection = (index) => {
		props.actions.selectItem({
			event: 'cliqz_adb_mode',
			value: index,
		});
	};

	return (
		<div className="s-tabs-panel">
			<div className="row">
				<div className="columns">
					<h3>{ t('settings_adblocker') }</h3>
					<h5>{ t('settings_adblocker_lists') }</h5>
					<RadioButtonGroup
						labels={['settings_adblocker_list_1', 'settings_adblocker_list_2', 'settings_adblocker_list_3']}
						handleItemClick={handleListSelection}
						indexClicked={settingsData.cliqz_adb_mode}
					/>
				</div>
			</div>
		</div>
	);
};

AdBlocker.propTypes = {
	actions: PropTypes.shape({
		selectItem: PropTypes.func.isRequired,
	}).isRequired,
	settingsData: PropTypes.shape({
		cliqz_adb_mode: PropTypes.number,
	}),
};

AdBlocker.defaultProps = {
	settingsData: {
		cliqz_adb_mode: 0,
	},
};

export default AdBlocker;
