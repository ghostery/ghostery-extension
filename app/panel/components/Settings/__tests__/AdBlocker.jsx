/**
 * AdBlocker Settings Test Component
 *
 * Ghostery Browser Extension
 * https://www.ghostery.com/
 *
 * Copyright 2020 Ghostery, Inc. All rights reserved.
 *
 * This Source Code Form is subject to the terms of the Mozilla Public
 * License, v. 2.0. If a copy of the MPL was not distributed with this
 * file, You can obtain one at http://mozilla.org/MPL/2.0
 */

import React from 'react';
import renderer from 'react-test-renderer';
import { mount } from 'enzyme';
import AdBlocker from '../AdBlocker';

describe('app/panel/Settings/AdBlocker.jsx', () => {
	describe('Snapshot tests with react-test-renderer', () => {
		test('AdBlocker is rendered correctly with AdOnly checked', () => {
			const settingsData = {
				cliqz_adb_mode: 0,
			};
			const actions = {
				selectItem: () => {},
			};

			const component = renderer.create(
				<AdBlocker
					settingsData={settingsData}
					actions={actions}
				/>
			).toJSON();
			expect(component).toMatchSnapshot();
		});

		test('AdBlocker is rendered correctly with Ads & Trackers checked', () => {
			const settingsData = {
				cliqz_adb_mode: 1,
			};
			const actions = {
				selectItem: () => {},
			};

			const component = renderer.create(
				<AdBlocker
					settingsData={settingsData}
					actions={actions}
				/>
			).toJSON();
			expect(component).toMatchSnapshot();
		});

		test('AdBlocker is rendered correctly with Ads, Trackers, & Annoyances checked', () => {
			const settingsData = {
				cliqz_adb_mode: 2,
			};
			const actions = {
				selectItem: () => {},
			};

			const component = renderer.create(
				<AdBlocker
					settingsData={settingsData}
					actions={actions}
				/>
			).toJSON();
			expect(component).toMatchSnapshot();
		});
	});

	describe('Shallow snapshot tests rendered with Enzyme', () => {
		test('AdBlocker functions correctly', () => {
			const settingsData = {
				cliqz_adb_mode: 0,
			};
			const actions = {
				selectItem: jest.fn(),
			};

			const component = mount(
				<AdBlocker
					settingsData={settingsData}
					actions={actions}
				/>
			);

			expect(actions.selectItem.mock.calls.length).toBe(0);
			component.find('.RadioButtonGroup__label').at(0).simulate('click');
			component.find('.RadioButton__outerCircle').at(2).simulate('click');
			expect(actions.selectItem.mock.calls.length).toBe(2);
		});
	});
});
