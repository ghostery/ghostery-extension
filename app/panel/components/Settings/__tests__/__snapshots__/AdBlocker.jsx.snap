// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`app/panel/Settings/AdBlocker.jsx Snapshot tests with react-test-renderer AdBlocker is rendered correctly with AdOnly checked 1`] = `
<div
  className="s-tabs-panel"
>
  <div
    className="row"
  >
    <div
      className="columns"
    >
      <h3>
        settings_adblocker
      </h3>
      <h5>
        settings_adblocker_lists
      </h5>
      <div
        className="RadioButtonGroup__container"
      >
        <div
          className="flex-container align-justify RadioButtonGroup__labelContainer"
        >
          <div
            className="RadioButtonGroup__label"
            onClick={[Function]}
          >
            settings_adblocker_list_1
          </div>
          <div
            className="RadioButtonGroup__label"
            onClick={[Function]}
          >
            settings_adblocker_list_2
          </div>
          <div
            className="RadioButtonGroup__label"
            onClick={[Function]}
          >
            settings_adblocker_list_3
          </div>
        </div>
        <div
          className="flex-container align-justify RadioButtonGroup__buttonsContainer"
        >
          <div
            className="RadioButtonGroup__button"
          >
            <span>
              <span
                className="RadioButton__outerCircle checked"
                onClick={[Function]}
              >
                <span
                  className="RadioButton__innerCircle checked"
                />
              </span>
            </span>
          </div>
          <div
            className="RadioButtonGroup__button"
          >
            <span>
              <span
                className="RadioButton__outerCircle"
                onClick={[Function]}
              >
                <span
                  className="RadioButton__innerCircle"
                />
              </span>
            </span>
          </div>
          <div
            className="RadioButtonGroup__button"
          >
            <span>
              <span
                className="RadioButton__outerCircle"
                onClick={[Function]}
              >
                <span
                  className="RadioButton__innerCircle"
                />
              </span>
            </span>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`app/panel/Settings/AdBlocker.jsx Snapshot tests with react-test-renderer AdBlocker is rendered correctly with Ads & Trackers checked 1`] = `
<div
  className="s-tabs-panel"
>
  <div
    className="row"
  >
    <div
      className="columns"
    >
      <h3>
        settings_adblocker
      </h3>
      <h5>
        settings_adblocker_lists
      </h5>
      <div
        className="RadioButtonGroup__container"
      >
        <div
          className="flex-container align-justify RadioButtonGroup__labelContainer"
        >
          <div
            className="RadioButtonGroup__label"
            onClick={[Function]}
          >
            settings_adblocker_list_1
          </div>
          <div
            className="RadioButtonGroup__label"
            onClick={[Function]}
          >
            settings_adblocker_list_2
          </div>
          <div
            className="RadioButtonGroup__label"
            onClick={[Function]}
          >
            settings_adblocker_list_3
          </div>
        </div>
        <div
          className="flex-container align-justify RadioButtonGroup__buttonsContainer"
        >
          <div
            className="RadioButtonGroup__button"
          >
            <span>
              <span
                className="RadioButton__outerCircle"
                onClick={[Function]}
              >
                <span
                  className="RadioButton__innerCircle"
                />
              </span>
            </span>
          </div>
          <div
            className="RadioButtonGroup__button"
          >
            <span>
              <span
                className="RadioButton__outerCircle checked"
                onClick={[Function]}
              >
                <span
                  className="RadioButton__innerCircle checked"
                />
              </span>
            </span>
          </div>
          <div
            className="RadioButtonGroup__button"
          >
            <span>
              <span
                className="RadioButton__outerCircle"
                onClick={[Function]}
              >
                <span
                  className="RadioButton__innerCircle"
                />
              </span>
            </span>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`app/panel/Settings/AdBlocker.jsx Snapshot tests with react-test-renderer AdBlocker is rendered correctly with Ads, Trackers, & Annoyances checked 1`] = `
<div
  className="s-tabs-panel"
>
  <div
    className="row"
  >
    <div
      className="columns"
    >
      <h3>
        settings_adblocker
      </h3>
      <h5>
        settings_adblocker_lists
      </h5>
      <div
        className="RadioButtonGroup__container"
      >
        <div
          className="flex-container align-justify RadioButtonGroup__labelContainer"
        >
          <div
            className="RadioButtonGroup__label"
            onClick={[Function]}
          >
            settings_adblocker_list_1
          </div>
          <div
            className="RadioButtonGroup__label"
            onClick={[Function]}
          >
            settings_adblocker_list_2
          </div>
          <div
            className="RadioButtonGroup__label"
            onClick={[Function]}
          >
            settings_adblocker_list_3
          </div>
        </div>
        <div
          className="flex-container align-justify RadioButtonGroup__buttonsContainer"
        >
          <div
            className="RadioButtonGroup__button"
          >
            <span>
              <span
                className="RadioButton__outerCircle"
                onClick={[Function]}
              >
                <span
                  className="RadioButton__innerCircle"
                />
              </span>
            </span>
          </div>
          <div
            className="RadioButtonGroup__button"
          >
            <span>
              <span
                className="RadioButton__outerCircle"
                onClick={[Function]}
              >
                <span
                  className="RadioButton__innerCircle"
                />
              </span>
            </span>
          </div>
          <div
            className="RadioButtonGroup__button"
          >
            <span>
              <span
                className="RadioButton__outerCircle checked"
                onClick={[Function]}
              >
                <span
                  className="RadioButton__innerCircle checked"
                />
              </span>
            </span>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
